internal/api/common.go

```diff
@@ -91,36 +93,42 @@ func EncodeResponse(_ context.Context, w http.ResponseWriter, response interface
 func EncodeError(_ context.Context, err error, w http.ResponseWriter) {
 	w.Header().Set("Content-Type", ContentType)
 	switch {
-	case errors.Contains(err, errors.ErrMalformedEntity),
-		err == apiutil.ErrMissingID,
-		err == apiutil.ErrEmptyList,
-		err == apiutil.ErrMissingMemberType,
-		errors.Contains(err, apiutil.ErrInvalidSecret),
-		err == apiutil.ErrNameSize:
+	case strings.Contains(err.Error(), apiutil.ErrMalformedEntity.Error()),
```

Comment:
I suspect that you used this because the order of the cases matters. However, this is not a solution but a workaround. Try to figure out a better solution (i.e. make error wraps order and cases order in such a way it makes sense).